src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1053,6 +1061,11 @@ protected Task ProduceEnd()
             }
```

Comment:
Take a closer look at the usages of HasResponseStarted. The one above should be changed to HasFlushedHeaders otherwise ProduceEnd() will try to modify already-commited response headers in a vain attempt to produce a 500 response.

Reply:
If you are referring to: 
```c#
                if (HasResponseStarted)
                {
                    // We can no longer change the response, so we simply close the connection.
                    _keepAlive = false;
                    OnErrorAfterResponseStarted();
                    return Task.CompletedTask;
                }
```

I think that check is correct. If HasResponseStarted or HasFlushedHeaders is true (which is always true if HasResponseStarted is true), we don't want to set headers. Changing it to HasFlushedHeaders would do the opposite and try to write headers after the response has already started.

Reply:
You're right. I thought HasResponseStarted meant HasFlushedHeaders. I think we should rename HasResponseStarted to HasCommittedHeaders internally to avoid this confusion. IHttpResponseFeature.HasStarted would still map to HasCommittedHeaders in this case.